### PR TITLE
Bump go.mod to use Go 1.18

### PR DESCRIPTION
**Pull request by reedloden**
We're now using Go 1.18 features (native fuzzing in https://github.com/gravitational/teleport/pull/13473), which means we need to accurately state that we require Go 1.18 in our go.mod.

### CODE REVIEW

**Review by jakule — APPROVED** [review]
@reedloden Just to confirm. You're bumping up the version only on master or you're also going to do it on v10?

**Comment by reedloden** [comment]
> @reedloden Just to confirm. You're bumping up the version only on master or you're also going to do it on v10?

I defer to the core team as to what you all would want. For my purposes for fuzzing, I am only caring about master. But might be worthwhile to do v10 as well? Let me know.

**Comment by zmb3** [comment]
We can keep the fuzz tests in master only for now, but let's put the 1.18 directive in v10 as well. 

This will prevent a new change that leverages 1.18 features from failing when it is backported

**Review by zmb3 — APPROVED** [review]
We should probably do the API module too.

File: api/go.mod
Comment (zmb3) [inline api/go.mod:27]: How did we get all these new dependencies?

File: go.mod
Comment (zmb3) [inline go.mod:127]: Where did this come from?

File: api/go.mod
Comment (reedloden) [reply api/go.mod:27]: `go mod tidy` added them.

File: go.mod
Comment (reedloden) [reply go.mod:127]: `go mod tidy`

Comment (reedloden) [reply go.mod:127]: This dependency was specifically added in commit aaa6ea73dd.

**Review by zmb3 — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

File: go.mod
Comment (greedy52) [reply go.mod:127]: i think you need to have `e` submodule checked out to run `go mod tidy`

**Pull request merged**
